source/SH3/graphics/msbmp.cpp

```diff
@@ -0,0 +1,115 @@
+/** @file
+ *
+ *  @copyright 2016  Palm Studios
+ *
+ *  @date 6-2-17
+ *
+ *  @author Jesse Buhagiar
+ */
+
+#include "SH3/graphics/msbmp.hpp"
+#include "SH3/system/log.hpp"
+
+#include <fstream>
+#include <algorithm>
+
+
+struct pixel
+{
+    std::uint8_t r, g, b;
+};
+
+using namespace sh3_graphics;
+
+msbmp::load_result msbmp::Load(const std::string& path)
+{
+    std::ifstream   file;
+    bmp_file_header fheader;    // File information header
+    bmp_info_header iheader;
+    load_result     res;
+    std::uint16_t   bpp;        // Bitmap bit depth
+
+    std::vector<std::uint32_t> palette;     // Vector containing colour data for this image
```

Comment:
It might use byte-sized moves instead of bit twiddling, unless the optimizer decides that the latter is more efficient on the platform (keeping [this](http://wiki.c2.com/?SufficientlySmartCompiler) in mind).
Using a struct results in less magic constants though.